huaweicloud/services/acceptance/smn/resource_huaweicloud_smn_subscription_test.go

```diff
@@ -5,35 +5,64 @@ import (
 	"testing"
 
 	"github.com/chnsz/golangsdk/openstack/smn/v2/subscriptions"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
 	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
 	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
 	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
 )
 
+func getResourceSMNSubscription(conf *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	smnClient, err := conf.SmnV2Client(acceptance.HW_REGION_NAME)
+	if err != nil {
+		return nil, fmtp.Errorf("error creating smn: %s", err)
+	}
+
+	var subscription *subscriptions.SubscriptionGet
+	foundList, err := subscriptions.List(smnClient).Extract()
```

Comment:
s/subscriptions.List/subscriptions.ListFromTopic(client, topicUrn)
from the console , actual: Subscription urn = {topic urn}:UUID